src/EditorFeatures/Core/Implementation/ReferenceHighlighting/AbstractDocumentHighlightsService.cs

```diff
@@ -223,7 +223,12 @@ private bool ShouldConsiderSymbol(ISymbol symbol)
                 list.Add(new DocumentHighlights(kvp.Key, spans.ToImmutableAndFree()));
             }
 
-            return list.ToImmutableAndFree();
+            var result = list.ToImmutableAndFree();
+
+            // Diagnosing flaky tests: assert that no span appears more than once
+            var groupedSpans = result.SelectMany(d => d.HighlightSpans).GroupBy(s => s);
+            Contract.ThrowIfTrue(groupedSpans.Any(s => s.Count() > 1), "Found duplicate spans!");
```

Comment:
(is this just for tracking down a test failure?  if so, that' sfine.  but it should not go into the product as a release-check).